apps/api/src/indexing/conviction-claims.repository.ts

```diff
@@ -0,0 +1,248 @@
+import { and, desc, eq } from "drizzle-orm";
+
+import { TxHashCoder } from "@recallnet/db/coders";
+import { convictionClaims } from "@recallnet/db/schema/conviction-claims/defs";
+
+import { db } from "@/database/db.js";
+
+/**
+ * ConvictionClaimsRepository
+ *
+ * Persistence layer for the conviction_claims table.
+ * This repository handles storage and retrieval of claim events from the conviction claims contract.
+ *
+ * Responsibilities:
+ * - Store claim events with idempotency checks
+ * - Query claims by account, season, or both
+ * - Track total claimed amounts per account
+ *
+ * Relationships:
+ * - Called by TransactionProcessor to store decoded claim transactions
+ * - Writes to `conviction_claims` table with blockchain metadata
+ *
+ * Conventions:
+ * - All addresses are stored lowercase
+ * - Transaction and block hashes are stored as bytea
+ * - Amounts are stored as bigint
+ */
+export class ConvictionClaimsRepository {
+  readonly #db: typeof db;
+
+  constructor(database: typeof db = db) {
+    this.#db = database;
+  }
+
+  /**
+   * Check if a claim event has already been recorded.
+   *
+   * Idempotency key: transaction_hash
+   *
+   * Returns:
+   * - true  → already present (caller should skip processing)
+   * - false → new event (caller may proceed)
+   */
+  async isConvictionClaimPresent(transactionHash: string): Promise<boolean> {
```

Comment:
I usually include an optional `txn?: Transaction` parameter in case the caller wants to wrap the call to any repository function in a transaction. You can see this pattern in other repository code in the `db` package. If it's not useful in this indexing context, disregard.